File: src/SampleJavaBasic/src/org/csystem/math/geometry/Point.java
/*----------------------------------------------------------------------------------------------------------------------
	Point sınıfı
----------------------------------------------------------------------------------------------------------------------*/
package org.csystem.math.geometry;

import static java.lang.Math.sqrt;

public class Point {
	public int x;
	public int y;

	Point(Point point)
	{
		x = point.x;
		y = point.y;
	}

	public Point()
	{		
	}
	
	public Point(int x)
	{
		this.x = x;
	}
	
	public Point(int x, int y)
	{
		this.x = x;
		this.y = y;
	}
	
	public double distance()
	{
		return distance(0);
	}

	public double distance(Point p)
	{
		return this.distance(p.x, p.y);
	}
	
	public double distance(int a)
	{
		return distance(a, 0);
	}

	public double distance(int a, int b)
	{
		return sqrt((x - a) * (x - a) + (y - b) * (y - b));								
	}
	
	public void offset(int dxy)
	{
		this.offset(dxy, dxy);
	}
	
	public void offset(int dx, int dy)
	{
		x += dx;
		y += dy;
	}
	
	public String toString()
	{
		return String.format("{x: %d, y: %d}", x, y);
	}	
}

File: src/Java-Kotlin/Sample/src/main/java/org/csystem/util/math/geometry/Point.kt
/*----------------------------------------------------------------------------------------------------------------------
    Point sınıfı
----------------------------------------------------------------------------------------------------------------------*/
package org.csystem.util.math.geometry

import kotlin.math.sqrt

data class Point(var x: Int = 0, var y: Int = 0) {
    fun distance(p: Point) = distance(p.x, p.y)

    fun distance(x: Int = 0, y: Int = 0)
            =  sqrt((this.x - x.toDouble()) * (this.x - x.toDouble()) + (this.y - y.toDouble()) * (this.y - y.toDouble()))

    fun offset(dx: Int, dy: Int = dx)
    {
        x += dx
        y += dy
    }
}